### ISSUE TITLE
[B039] False-positive on mutable default when ContextVar has immutable type

### ISSUE THREAD

**Issue by brandonchinn178** [astral-sh/ruff#20696]
### Summary

The following code fails linting:
```python
FOO: ContextVar[Sequence[str]] = ContextVar("foo", default=[])
```
But this should be safe because call-sites cannot mutate it since it's a `Sequence`

**Comment by ntBre** [comment]
Thanks for the report!

I'm not totally sure if we should consider this a false positive. Not being able to append to the `Sequence` seems like an invariant that only a type-checker could enforce, and at runtime it still seems possible to mutate the shared copy:

```pycon
>>> from contextvars import ContextVar
>>> from collections.abc import Sequence
>>> FOO: ContextVar[Sequence[str]] = ContextVar("foo", default=[])
>>>
>>> FOO.get()
[]
>>> FOO.get().append(1)
>>> FOO.get()
[1]
>>>
```

But I'm certainly open to other opinions!

**Comment by brandonchinn178** [comment]
But isn't that always true? For example, ruff allows the following with B006:

```py
def foo (x: Sequence[int] = [])
```

**Comment by ntBre** [comment]
Oh you're right. We already have an `is_immutable_annotation` that covers `Sequence` and that we use in B006. Maybe we should be using that here as well.
